lib/testing/test_runner.sh

```diff
@@ -69,6 +69,18 @@ OUTSIDE_MAC=
 ENABLE_IPV4=0
 ENABLE_VLAN=0
 
+version_greater_or_equal()
+{
+    printf '%s\n%s\n' "$2" "$1" | sort -V -C
+}
+
+assert_kernel_version_greater_or_equal()
+{
+    if ! version_greater_or_equal "$(uname -r)" "$1"; then
```

Comment:
Did you test this on RHEL? ;)

We have a... creative... relationship with version numbers, and others backport features too, so I'm afraid a version number-based test is not going to cut it :(

An alternative would be to have each test detect the log output that indicates a compatibility error and turn a FAIL into SKIP for that. Maybe have each test define a "skip regex" that is automatically applied to the output by the test framework?

Reply:
Or just always grep for the "compatibility check failed" log output line...